src/python_testing/TC_ICDM_2_1.py

```diff
@@ -194,7 +194,7 @@ async def test_TC_ICDM_2_1(self):
 
         # Validate ClientsSupportedPerFabric
         self.step(5)
-        if self.pics_guard(self.check_pics("ICDM.S.A0005")):
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.ClientsSupportedPerFabric):
```

Comment:
thinking it makes more sense to have this be a non-async function. See _async_runner for the wrapper on how to do that. I don't think you're ever going to want this to be truly async